src/mono/System.Private.CoreLib/src/System/Threading/Thread.Mono.Unix.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+using System.Runtime;
+using System.Runtime.InteropServices;
+
+namespace System.Threading
+{
+    public sealed partial class Thread
```

Comment:
This is not really Mono specific, wouldn't be better to have it in shared part under WaitSubsystem?

Reply:
We need to map those two methods to `WaitSubsystem`, but CoreCLR is still using their separate implementation for now so the shared portion isn't an option. If they adopt `WaitSubsystem`, then this (along with a variety of other stuff) could get moved to shared.

Reply:
My suggestion was to conditional it in shared in same way as is the whole WaitSubsystem. The CoreCLR adoption is not a separate thing.